### PR TITLE
GH1053: Added support for custom #load schemes

### LINKED ISSUE #1053: Feature request downloadable .cake script directive
I need to use the #load but its not working with esey when i host it with nuget as it needs a path so i created the #nuscript directive.

hope this can get resolved and even added if its all done ok.

### PR DESCRIPTION
**Pull request by patriksvensson**
Also adds support for loading scripts from NuGet packages.

Closes #1053

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -46,6 +46,7 @@ Setup(context =>
 //////////////////////////////////////////////////////////////////////
 
 Task("Clean")
+    .WithCriteria(() => false)
```
Comment (gep13) [inline build.cake]: @patriksvensson don't think this is needed anymore 😄

File: src/stylecop.json [hunk 2]
```diff
@@ -11,13 +11,13 @@
           "documentPrivateElements": false,
           "documentInterfaces": true,
           "documentPrivateFields": false,
+          "xmlHeader": false,
```
Comment (gep13) [inline src/stylecop.json]: @patriksvensson should this be part of a PR for the other issue?

**Comment by tlycken** [comment]
Nice - I've been looking forward to this!

What's the syntax to make use of it? Is it documented somewhere? (Couldn't find any docs chagnes in the commit history here...)

**Comment by gep13** [comment]
>@tlycken said...
What's the syntax to make use of it? Is it documented somewhere? 

That is something that will make it's way onto the website.  Follow this issue [here](https://github.com/cake-build/website/issues/232).  Although this PR is closed, the milestone that this is included in has not been released yet, so you won't be able to make use of it, unless you are pulling Cake from our MyGet feed.

Having said that, it follows the same conventions as you can see here:

http://cakebuild.net/docs/fundamentals/preprocessor-directives

i.e. you should be able to do:

`#load nuget:?package=Cake.Recipe`

where Cake.Recipe is the name of the package that you want to load.

**Pull request merged**